common/include/pcl/pcl_macros.h

```diff
@@ -55,8 +55,10 @@
   #pragma warning (disable: 4018 4244 4267 4521 4251 4661 4305 4503 4146)
 #endif
 
-#ifndef _USE_MATH_DEFINES
-#define _USE_MATH_DEFINES
+#if !(defined _WIN32 && (defined _MSC_VER || defined __MINGW32__))
```

Comment:
I'm a bit confused, so please humor me. Is this needed? What would be the issue if _USE_MATH_DEFINES is used on Windows?

Reply:
I did it that way since it seems like the combination 

```
#define _USE_MATH_DEFINES
#include <cmath>
```

doesn't define `_MATH_DEFINES_DEFINED` when I compile using mingw, but it *does* define `M_PI` and similar macros, because I get a lot of warnings about `warning: "M_PI" redefined` etc. So I included the test you commented since we want to force the manual definitions of `M_PI` etc. in those cases. 

Although when I think about it it's kind of weird that I get errors about redefinitions, while at the same time I get errors about `M_PI` not defined when I don't define them manually...

Reply:
I think the issue as you commented early on is that somehow cmath is included without this macro defined before. Solution might involve moving the math headers below the pcl headers and stuff like that

Reply:
That's a good idea, I'll check that!